indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/PartialSegmentGenerateTask.java

```diff
@@ -164,7 +171,11 @@ abstract T createGeneratedPartitionsReport(
     final PartitionsSpec partitionsSpec = tuningConfig.getGivenOrDefaultPartitionsSpec();
     final long pushTimeout = tuningConfig.getPushTimeout();
 
-    final IndexTaskSegmentAllocator segmentAllocator = createSegmentAllocator(toolbox);
+    final CachingSegmentAllocator segmentAllocator = createSegmentAllocator(toolbox, taskClient);
+    final SequenceNameFunction sequenceNameFunction = new NonLinearlyPartitionedSequenceNameFunction(
```

Comment:
`PartialSegmentGenerateTask` is used when shuffling intermediate segment is required. The hash or range partitionings need the shuffle, but for the linear partitioning, I think it would never need it.

Hmm, well actually I think I can make `SequenceNameFunction` to be more general so that it can create the proper sequence name from the partialShardSpec. I will do it in the follow-up pr.